controllers/roles.go

```diff
@@ -41,36 +43,50 @@ func (rc *roleController) UpdateRole(ctx echo.Context) error {
 
 	}
 	if err := rc.db.UpdateRole(&role); err != nil {
+		rc.LogInfo("Failed to update role: " + err.Error())
 		return ctx.JSON(http.StatusInternalServerError, echo.Map{"error": err.Error()})
 
 	}
+	rc.LogInfo("Role updated: " + role.Name)
 	return ctx.JSON(http.StatusOK, echo.Map{"role": role})
 }
 
 func (rc *roleController) DeleteRole(ctx echo.Context) error {
 	id, _ := strconv.ParseUint(ctx.Param("id"), 10, 64)
 	if err := rc.db.DeleteRole(uint(id)); err != nil {
+		rc.LogInfo("Failed to delete role: " + err.Error())
 		return ctx.JSON(http.StatusInternalServerError, echo.Map{"error": err.Error()})
 
 	}
+	rc.LogInfo("Role deleted: ID " + strconv.FormatUint(id, 10))
 	return ctx.JSON(http.StatusOK, echo.Map{"message": "Role deleted"})
 }
 
 func (rc *roleController) GetRole(ctx echo.Context) error {
 	id, _ := strconv.ParseUint(ctx.Param("id"), 10, 64)
 	role, err := rc.db.GetRole(uint(id))
 	if err != nil {
+		rc.LogInfo("Failed to get role: " + err.Error())
 		return ctx.JSON(http.StatusInternalServerError, echo.Map{"error": err.Error()})
 
 	}
+	rc.LogInfo("Role retrieved: ID " + strconv.FormatUint(id, 10))
```

Comment:
`zap.Fields` works well exactly here, for not concatenating information the hard way.

Reply:
👍👍